dts/bindings/pwm/ti,mspm0-pwm.yaml

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2024 Bang & Olufsen A/S, Denmark
+#
+# SPDX-License-Identifier: LicenseRef-BnOProprietary
+
+description: |
+  Driver to enable MSPM0 PWM hardware for TIMx timers (TIMA/TIMG).
+  This driver is implementing the existing Zephyr PWM API (pwm_driver_api) and
+  supports the set_cycles and get_cycles_per_sec functions. 
+  
+  For the set_cycles function, passing flags = 1 as the last argument will stop the running counter.
+  By default, making any call with flags = 0, will restart the counter
+
+compatible: "ti,mspm0-pwm"
+
+include: [base.yaml,pinctrl-device.yaml]
+
+properties:
+  prescaler:
+    type: int
+    default: 0
+    description: Specifies the value used to divide the input clock frequency, allowing adjustment of the effective clock rate used by the timer.
+
+  mode:
+    type: int
+    default: 0
+    description: Determines the operating mode of the timer.
+
+  clock-frequency:
+    type: int
+    required: true
+    description: Defines the clock frequency that the module will use. This value is required and should be treated as read-only; it must not be modified after initial configuration.
+
+  divide-ratio:
+    type: int
+    required: true
+    description: Sets the division ratio for the clock, which controls how the clock frequency is divided before use.
+
+  four-cc:
+    type: boolean
+    required: false
+    description: Timer specific. Defines whether the timer module can support 4 capture compare channels.
+
+  pinctrl-0:
+    required: true
+    description: References the pin control configuration necessary for the proper functioning of the timer.
+
+  is_tima:
+    type: boolean
+    required: false
+    description: Indicates whether the configuration is related to the TIMA or TIMG timer module.
+
+  cc_index:
+    type: int
+    required: true
+    description: Specifies the index of the capture/compare channel used by the timer.
+
+  direction:
+    type: int
+    required: true
+    description: Specifies the direction for a specific capture compare channel.
```

Comment:
Isn't this really a bitmask to configure all of the channels? Anyway, I think the bindings need to be revisited.

Reply:
removed